src/Controls/tests/TestCases.HostApp/Issues/Issue24489.xaml.cs

```diff
@@ -0,0 +1,92 @@
+using Microsoft.Maui.Controls.Shapes;
+
+namespace Maui.Controls.Sample.Issues;
+
+[Issue(IssueTracker.Github, 24489, "TitleBar Implementation", PlatformAffected.macOS)]
+
+public partial class Issue24489 : ContentPage
+{
+	TitleBar _customTitleBar;
+
+	public Issue24489()
```

Comment:
Would be nice to modify the sample to:
- For example, change one of the TitleBar labels by a Button to verify event handlers from the TitleBar.
- Tapping the added Button, modify the Layout to verify that invalidate and updates correctly the TitleBar size.